plugins/plugin-bitbucket/codenvy-plugin-bitbucketserver-webhooks/src/main/java/com/codenvy/plugin/webhooks/bitbucketserver/BitbucketServerWebhookService.java

```
@@ -157,34 +164,24 @@ private String computeCloneUrl(String owner, String projectKey, String repositor
     }
 
     private Set<String> getFactoriesIDs(final String repositoryUrl) throws ServerException {
-        Optional<BitbucketServerWebhook> optional = getConfiguredWebhooks().stream()
-                                                                           .filter(webhook -> webhook.getRepositoryCloneUrl()
-                                                                                                     .equals(repositoryUrl))
-                                                                           .findFirst();
-        if (optional.isPresent()) {
-            return optional.get().getFactoriesIds();
-        } else {
-            return emptySet();
+        Map<String, String> properties = configurationProperties.getProperties(WEBHOOK_PROPERTY_PATTERN);
+
+        Set<String> webhooks = properties.entrySet()
+                                         .stream()
+                                         .filter(entry -> repositoryUrl.equals(entry.getValue()))
+                                         .map(entry -> entry.getKey()
+                                                            .substring(0, entry.getKey().lastIndexOf(WEBHOOK_REPOSITORY_URL_SUFFIX)))
+                                         .collect(toSet());
+
+        if (webhooks.isEmpty()) {
+            LOG.error("No BitBucket Server webhooks were registered for repository {}", repositoryUrl);
```

If it is an error, should we throw an exception ?